### DISCUSSION TITLE
External Authorization

### DISCUSSION THREAD

**Discussion by wthornton18** [meilisearch/product#712] [Feedback & Feature Proposal]
It would be very useful for my use case to add an ability to delegate what documents a particular user should be allowed to access in an external Authorization system. 

With my current approach, I either have to temporarily generate access tokens with appropriate filters for a given user and then delete them after, or add a proxy to filter the results before returning them.

I imagine this could be implemented in the style of:
https://kubernetes.io/docs/reference/access-authn-authz/webhook/

Would be good to hear everyone's thoughts

**Comment by wthornton18** [comment]
Just to qualify, I would be more than to help build such a feature if it was something that was considered useful.

**Comment by macraig** [comment]
Hello @wthornton18, thank you for taking the time to share your feedback. While we greatly value your suggestion, incorporating this feature isn't on our current development roadmap. As part of our commitment to maintaining the Meilisearch's security standards, we're not accepting external contributions for a feature like this at this time.

However, we'll keep this discussion open in case other users wish to share their thoughts or find it beneficial to the Meilisearch's evolution. We appreciate your understanding and continued support!
